pages/common/xargs.md

```diff
@@ -1,19 +1,20 @@
 # xargs
 
-> Execute a command with piped arguments.
+> Execute a command with piped arguments coming from another command, a file, etc.
+> The input is treated as a single block of text and split into separate arguments on spaces, tabs, newlines and end-of-file.
 
-- Main use:
+- Main usage pattern:
 
-`{{arguments}} | xargs {{command}}`
+`{{arguments_source}} | xargs {{command}}`
 
-- Specific example: delete all files that start with 'M':
+- Delete all files with a `.backup` extension:
 
-`find . -name 'M*' | xargs rm`
+`{{find . -name '*.backup'}} | xargs {{rm -v}}`
 
-- Handle whitespace in arguments:
+- Use the NUL (`\0`) character as separator (useful to split the input on newlines but not on spaces):
```

Comment:
Also, both workarounds would be better served by the -d option, as pointed out by @jrziviani in #421, but that option does not exist in the version of xargs shipped with OSX, so we'd have to split the page, which sucks but may be the cleanest approach (see [osx/tree.md](../blob/master/pages/osx/tree.md) vs [linux/tree.md](../blob/master/pages/linux/tree.md), particularly in light of #990).